### ISSUE TITLE
Improve Linux support

### ISSUE THREAD

**Issue by GeckoEidechse** [R2Northstar/Northstar#214]
# Introduction

Since the SteamDeck verified program has started, vanilla Titanfall 2 has been marked as **_Playable_**, meaning the game works out-of-the-box on Linux!

![image](https://user-images.githubusercontent.com/40122905/158029228-5f43a696-52da-440b-a622-19f06070e801.png)

The same however cannot be said about Northstar. While we feature a [dedicated section on the Wiki](https://r2northstar.gitbook.io/r2northstar-wiki/using-northstar/playing-on-linux) that contains instructions for installing on Linux and describes various workarounds to known issues, running Northstar on Linux is generally more difficult than on Windows.

## Why improved Linux support?

As per [Steam Survey](https://store.steampowered.com/hwsurvey), only slightly above 1% of players are on Linux, compared to the 96% on Windows. As such Windows will stay our main target. However with the introduction of the SteamDeck and improved support of running Windows native games on Linux, the relative number of players is only expected to rise (though never reaching a majority).
As such we should try to ensure that running Northstar on Linux is not significantly more difficult than compared to Windows or compared to running base Titanfall 2 on Linux.

# Issues and paper-cuts

While the initial installation (i.e. copying files to the Titanfall 2 directory) is the same for all OS's and platforms, Linux users have to make adjustments to their game depending on whether they are using [Steam](https://r2northstar.gitbook.io/r2northstar-wiki/using-northstar/playing-on-linux#steam-proton) or [Origin](https://r2northstar.gitbook.io/r2northstar-wiki/using-northstar/playing-on-linux#lutris-wine).

- Steam users for example will have to trick Steam into running `NorthstarLauncher.exe` by renaming `Titanfall2.exe` and then renaming or symlinking `NorthstarLauncher.exe` to `Titanfall2.exe`. This makes switching between the vanilla game and Northstar more difficult.

- The game console will out-of-the-box not show any text on Linux due to using a (by default) unsupported font.

- Some users have reported [needing to use specific versions of proton in order to be able to run Northstar](https://github.com/R2Northstar/Northstar/issues/1#issuecomment-1062483190) (while the base game works with default versions).

### Notes:

- Due to their native Linux support for the client and containerised environment, the focus should be on the Steam version of Linux. For example, the easiest solution would be if we could support adding the `-northstar` to Steam launch arguments also on Linux.
- [As per the wiki](https://r2northstar.gitbook.io/r2northstar-wiki/using-northstar/playing-on-linux) we say that we do not "officially" support Linux, Considering that some of our contributors run Northstar (exclusively) on Linux we might want to update that statement, should some of the issue stated above be resolved.
- [Viper](https://github.com/0neGal/viper) is the only Northstar installer and mod manager known to also have a native Linux version. It's main developer is working on [launching Northstar on Linux directly via Viper](https://github.com/0neGal/viper/issues/38).

### Related issues:

#1, #7, #53, #147, and #210

**Comment by pg9182** [comment]
> Due to their native Linux support for the client and containerised environment, the focus should be on the Steam version of Linux. For example, the easiest solution would be if we could support adding the -northstar to Steam launch arguments also on Linux.

I don't think this is necessary; tf2 works well out-of-the-box on pretty much any wine 6.x or newer.

**Comment by abarichello** [comment]
> * [As per the wiki](https://r2northstar.gitbook.io/r2northstar-wiki/using-northstar/playing-on-linux) we say that we do not "officially" support Linux, Considering that some of our contributors run Northstar (exclusively) on Linux we might want to update that statement, should some of the issue stated above be resolved.

Changing that right now

**Comment by IDeathByte** [comment]
It possible to get origin token without origin - through Northstar launcher options?
If i launch origin through lutris through steam (yep, that ugly way for make steam overlay works with origin), it can crash steam after few times, and steam controller stopped works with game.
Or mouse input can start jittering after any time, cause worse integration (if play without steam, all fine).

**Comment by WilsonBradley** [comment]
Agreen, keeping Origin tool from popping up would be nice.

**Comment by GeckoEidechse** [comment]
> It possible to get origin token without origin - through Northstar launcher options?

That would require Northstar handling your Origin credentials which is something I personally wouldn't want to have to implement and maintain ._.

My personal recommendation for anyone on Linux is to put the Steam version of Titanfall2 on your wishlist and re-buy it, the moment it's <10€.
I'm well aware that the Steam version still uses Origin in the back and requires you to log into Origin but for some reason Steam seems to handle that quite smoothly.

**Comment by CTalvio** [comment]
I've been suffering from the Access Violation DEP bug, and no matter what I did I was unable to get the game to launch with Northstar. I'm using the Origin version via lutris, and my attempts to use the recommended version of Proton GE in lutris were unsuccessful.

Nothing worked. Until I tried deleting the LatencyFlex .dlls. It immediately launched on the next attempt, no other changes needed. Not even restarting Origin.

**Comment by GeckoEidechse** [comment]
@KyleGospo might be worth taking a look as you did the initial LatencyFlex PR

**Comment by KyleGospo** [comment]
Looking at this now, I'll update you both

**Comment by KyleGospo** [comment]
https://github.com/R2Northstar/NorthstarLauncher/pull/168

PR to potentially fix this, LatencyFleX received some pretty big updates recently.

**Comment by KyleGospo** [comment]
Above code worked for me in multiple versions of Proton-GE with multiple versions of LatencyFleX. @CTalvio after this is merged if you have any further issues please message me on Discord, you'll find me under contributors as "1/4 Life". There's a lot of room for user error with LatencyFleX since it's a manual install.

**Comment by BotchedRPR** [comment]
I've been chatting on Discord about this problem, but it seems like Northstar's DXVK DLLs are broken. The game crashes on Linux if you don't update them. For someone that packages Northstar - you might want to look into getting the latest DXVKs for 64bit (https://github.com/doitsujin/dxvk/) and packaging them inside the bin/x64_retail folder. If someone needs my system info:
CPU - AMD Ryzen 5 3600
GPU - NV GTX 1080ti
16GB DDR4 RAM
Pop_OS 22.04

**Comment by GeckoEidechse** [comment]
> I've been chatting on Discord about this problem, but it seems like Northstar's DXVK DLLs are broken (for linux). The game crashes on Linux if you don't update them. For someone that packages Northstar - you might want to look into getting the latest DXVKs for 64bit (https://github.com/doitsujin/dxvk/) and packaging them inside the bin/x64_retail folder.

To update based on the discussion on Discord we figured out that the DXVK dlls are downloaded by Steam and not actually shipped by Northstar.

c.f.:
- https://github.com/doitsujin/dxvk/issues/2658#issuecomment-1140439188
- https://discord.com/channels/920776187884732556/972909610140839956/980435227405062184

**Comment by BotchedRPR** [comment]
I feel so stupid now :)
Thanks everyone for the help anyways.

**Comment by KyleGospo** [comment]
@CTalvio that LatencyFleX change is in the latest version if you want to try it out

**Comment by KyleGospo** [comment]
If you have bottles, can you check if their proton based soda runner has these patches?

**Comment by cyrv6737** [comment]
How does the Soda runner differ from Proton? I am unfamiliar with that particular runner from Bottles and it's not documented in their Docs. 

Proton outside of steam is very unsupported, so even if it did work I don't see how that would help. Especially since the goal here is to get N* running out of the box on the version of Proton the Deck ships with.

If I find myself having the time (and patience to deal with Steam under wine) I can take a look regardless.

**Comment by KyleGospo** [comment]
@cyrv6737 https://twitter.com/usebottles/status/1545488730806534144

If it's missing those patches I'll work with them to correct that, we'd like to start recommending Bottles for Northstar since we can export a ready-made bottle with everything needed.

**Comment by cyrv6737** [comment]
@pg9182 Not that I noticed no but I didn't play that long. Did you notice the performance issues on `wine` or `wine-staging`?

**Comment by pg9182** [comment]
`wine`, haven't tried staging
